storage/api/Storage/Storage.cs

```diff
@@ -762,6 +782,184 @@ bool GetRequesterPays(string bucketName)
+            storage.LockBucketRetentionPolicy(bucketName,
+                bucket.Metageneration.Value);
+            bucket = storage.GetBucket(bucketName);
+            Console.WriteLine($"Retention policy for {bucketName} is now locked");
+            Console.WriteLine($"Retention policy effective as of {bucket.RetentionPolicy.EffectiveTime}");
+        }
+        // [END storage_lock_retention_policy]
+
+        // [START storage_remove_retention_policy]
+        private void RemoveBucketRetentionPolicy(string bucketName)
+        {
+            var storage = StorageClient.Create();
+            var bucket = storage.GetBucket(bucketName);
+            if (bucket.RetentionPolicy != null)
+            {
+                bool? isLockedOrNull = bucket?.RetentionPolicy.IsLocked;
+                bool isLocked =
+                    isLockedOrNull.HasValue ? isLockedOrNull.Value : false;
+                if (isLocked)
+                {
+                    throw new Exception("Retention Policy is locked.");
+                }
+
+                bucket.RetentionPolicy.RetentionPeriod = null;
+                bucket = storage.UpdateBucket(bucket, new UpdateBucketOptions()
+                {
+                    IfMetagenerationMatch = bucket.Metageneration
+                });
+
+                Console.WriteLine($"Retention period for {bucketName} has been removed.");
+            }
+        }
+        // [END storage_remove_retention_policy]
+
+        // [START storage_get_retention_policy]
+        private void GetBucketRetentionPolicy(string bucketName)
+        {
+            var storage = StorageClient.Create();
+            var bucket = storage.GetBucket(bucketName);
+
+            if (bucket.RetentionPolicy != null)
+            {
+                Console.WriteLine("Retention policy:");
+                Console.WriteLine($"period: {bucket.RetentionPolicy.RetentionPeriod}");
+                Console.WriteLine($"effective time: {bucket.RetentionPolicy.EffectiveTime}");
+                bool? isLockedOrNull = bucket?.RetentionPolicy.IsLocked;
+                bool isLocked =
+                    isLockedOrNull.HasValue ? isLockedOrNull.Value : false;
+                Console.WriteLine("policy locked: {0}", isLocked);
+            }
+        }
+        // [END storage_get_retention_policy]
+
+        // [START storage_enable_default_event_based_hold]
+        private void EnableBucketDefaultEventBasedHold(string bucketName)
+        {
+            var storage = StorageClient.Create();
+            var bucket = storage.GetBucket(bucketName);
+            bucket.DefaultEventBasedHold = true;
+            bucket = storage.UpdateBucket(bucket, new UpdateBucketOptions()
+            {
+                // Use IfMetagenerationMatch to avoid race conditions.
+                IfMetagenerationMatch = bucket.Metageneration
+            });
+            Console.WriteLine($"Default event-based hold was enabled for #{bucketName}");

```

Comment:
#{bucketName}

Consider removing the "#" symbol.

Reply:
Whoops, My Ruby got into my C#.